shared/src/domain-logic/opportunityFields.ts

```diff
@@ -10,16 +10,55 @@ export const OPPORTUNITY_FIELDS: (keyof ProfileData)[] = [
 
 export const REPORT_FIELDS: (keyof ProfileData)[] = ["dateOfFormation"];
 
+export const FIELDS_FOR_FOREIGN_NEXUS_NJ_LOCATION: (keyof ProfileData)[] = ["municipality"];
+export const FIELDS_FOR_FOREIGN_NEXUS_NOT_NJ_LOCATION: (keyof ProfileData)[] = ["homeBasedBusiness"];
+export const ALL_FIELDS_FOR_FOREIGN_NEXUS_NJ_LOCATION = [
+  ...FIELDS_FOR_FOREIGN_NEXUS_NJ_LOCATION,
+  ...REPORT_FIELDS,
+];
+export const ALL_FIELDS_FOR_FOREIGN_NEXUS_NOT_NJ_LOCATION = [
+  ...FIELDS_FOR_FOREIGN_NEXUS_NOT_NJ_LOCATION,
+  ...REPORT_FIELDS,
+];
 export const FIELDS_FOR_PROFILE = [...REPORT_FIELDS, ...OPPORTUNITY_FIELDS];
-export const getFieldsForProfile = (legalStructureId?: string): (keyof ProfileData)[] => {
-  return FIELDS_FOR_PROFILE.filter((field) => {
+
+export const getFieldsForProfile = (profileData: ProfileData): (keyof ProfileData)[] => {
+  if (profileData.businessPersona === "FOREIGN" && profileData.foreignBusinessType === "NEXUS") {
+    return profileData.nexusLocationInNewJersey
+      ? lookupByLegalStructure(ALL_FIELDS_FOR_FOREIGN_NEXUS_NJ_LOCATION, profileData.legalStructureId)
+      : lookupByLegalStructure(ALL_FIELDS_FOR_FOREIGN_NEXUS_NOT_NJ_LOCATION, profileData.legalStructureId);
+  } else {
+    return lookupByLegalStructure(FIELDS_FOR_PROFILE, profileData.legalStructureId);
+  }
+};
+
+const lookupByLegalStructure = (
+  key: (keyof ProfileData)[],
```

Comment:
[pebble] Variable naming conventions - if something is an array, it should always be named in plural. This isn't `key` it is `keys` - or even better, to be truly descriptive - (`keys` begs the question "what keys?") - you could go with `profileKeys`

Reply:
changed to profileKeys